apps/web/src/components/layout/EnsureOnboardingComplete.tsx

```diff
@@ -0,0 +1,15 @@
+import { Navigate, useLocation } from 'react-router-dom';
+import { ROUTES } from '../../constants/routes.enum';
+import { useBlueprint, useAuthController } from '../../hooks/index';
+
+export function EnsureOnboardingComplete({ children }: any) {
+  useBlueprint();
+  const location = useLocation();
+  const { user } = useAuthController();
+
+  if ((!user?.organizationId || !user?.environmentId) && location.pathname !== ROUTES.AUTH_APPLICATION) {
+    return <Navigate to={ROUTES.AUTH_APPLICATION} replace />;
+  } else {
+    return children;
+  }
+}
```

Comment:
This file's contents came from `RequiredAuth`, and is heavily simplified (expand diff below). The `useBlueprint` hook call here is uncomfortable, but I've left it to avoid further scope in this PR.